### DISCUSSION TITLE
Proper way to make a cache persitent in DIRAC

### DISCUSSION THREAD

**Discussion by cserf** [DIRACGrid/DIRAC#6009] [Q&A]
Hi,

I have a question relative to cache. Here is the issue : Every time a `RucioFileCatalogClient` is instantiated it needs to get a list of valid scopes from Rucio : https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Resources/Catalog/RucioFileCatalogClient.py#L129 . It leads to a few 100 ms up to 1s of latency depending on RTT. This list of scopes is something almost static and can be cached. For agents or services `DIRAC.Core.Utilities.DictCache` is something that can be used, but for any CLI method calling the file catalog it won't work since the cache is not persistent between two calls.
I have a poor man solution based on pickles, but pickles is evil... I was wondering if there's any possibility to have a persistent cache using DIRAC ?

Cedric

**Comment by chaen** [comment]
If I am not mistaken, there is an agent which syncs some configuration between Rucio and DIRAC. Why not putting this almost static list in the CS ? 
Or if you are worried about `pickle`, use `json`

**Comment by fstagni** [comment]
Maybe try (inspiration from) https://github.com/DIRACGrid/COMDIRAC/blob/integration/src/COMDIRAC/Interfaces/Utilities/DConfigCache.py

**Comment by fstagni** [comment]
I believe that this could be closed. We do not have anyway more info to supply.
